src/pani/core.cljs

```diff
@@ -0,0 +1,161 @@
+(ns pani.core
+  (:refer-clojure :exclude [get-in set! reset! conj! swap! deref parents key])
+  (:require cljsjs.firebase
+            [clojure.string :as str]
+            [pani.registry :refer [register-listener]]))
+
+;; constants
+
+(def child-events
+  [:child-added
+   :child-changed
+   :child-moved
+   :child-removed])
+
+(def all-events
+  (conj child-events :value))
+
+;; utils
+
+;; FIXME: camel-case keys?
+;;        hydrate to custom vectors to preserve rich keys?
+;;        preserve sets (don't coerce to vector)
+;;        similarly preserve keywords as values
+
+(defn- hydrate [js-val]
+  (js->clj js-val :keywordize-keys true))
+
+(defn- serialize [clj-val]
+  (clj->js clj-val))
```

Comment:
I wonder if it might make sense to just use  `clj->js` everywhere. I see that this way the behavior can be modified more easily but I can't see a reason this would happen inside this lib.